### PR TITLE
Add string and bytes buffer convert trick in README

### LINKED ISSUE #1031: Consider using copy-less []byte/string conversion
Many of the optimizations in fasthttp are based on using byte arrays instead of strings. That is wise!

However, there is an assumption that translating these into strings (or back) requires a copy and should thus be discouraged.

With some caveats, this is not true. There are in fact reliable ways to avoid copying. Please see the code I use, [here](https://github.com/tliron/kutil/blob/main/util/string.go). I urge you to consider these or similar functions to handle conversion, and perhaps to benchmark the memory use and total throughput accordingly.

(The major caveat is the use of "unsafe". Some projects decide, as a matter of principle, to not use it. However, for the same reason fasthttp doesn't try to copy Go's http library, I believe that "unsafe" would be acceptable here.)

For completion, here are some references to code and discussion on this topic:

* https://go101.org/article/unsafe.html
* https://github.com/golang/go/issues/25484
* https://github.com/golang/go/issues/19367
* https://golang.org/src/strings/builder.go#L45

### PR DESCRIPTION
**Pull request by ichxxx**
close #1031

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -485,6 +485,28 @@ statusCode, body, err := fasthttp.Get(nil, "http://google.com/")
 uintBuf := fasthttp.AppendUint(nil, 1234)
 ```
 
+* String and `[]byte` buffer may converts wihtout memory allocation
```
Comment (erikdubbelboer) [inline README.md:488]:
```suggestion
* String and `[]byte` buffers may converted without memory allocations
```

File: README.md [hunk 2]
```diff
@@ -485,6 +485,28 @@ statusCode, body, err := fasthttp.Get(nil, "http://google.com/")
...
+    bh := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+    sh := (*reflect.StringHeader)(unsafe.Pointer(&s))
+    bh.Data = sh.Data
+    bh.Cap = sh.Len
+    bh.Len = sh.Len
+    return b
+}
+```
+The underlying structure of `[]byte` buffer has only one `Len` field more than String.
+So, we can construct one directly from the other.
```
Comment (erikdubbelboer) [inline README.md:504]: I'm not sure if it's useful to mention here.

File: README.md [hunk 3]
```diff
@@ -485,6 +485,28 @@ statusCode, body, err := fasthttp.Get(nil, "http://google.com/")
...
+
+Note:
+This is an unsafe way, 
+please make sure not to modify the bytes in the `[]byte` buffer if the result string still survives.
```
Comment (erikdubbelboer) [inline README.md:508]: I would like to see this warning bigger and bold as it's really important! It's not just a note.

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (ichxxx) [reply README.md:504]: I think some people are wondering how it works, but if people want to know more details, they'll Google it. It's ok to delete.

File: README.md (hunk 3 above)
Comment (ichxxx) [reply README.md]: Updated

**Comment by erikdubbelboer** [comment]
Thanks!

**Pull request merged**